### ISSUE TITLE
Rust for powerpc

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#451]
Meta issue tracking the state of Rust support in the kernel for powerpc.

# Basic support
 - [x] ppc64le supported in Rust-for-Linux/rust
 - [ ] ppc64 support ?
 - [ ] ppc32 support ?

# Upstream support
 - [ ] Rust build support in Github Actions for snowpatch
 - [ ] Rust build support in mpe's CI scripts
 - [ ] Rust build support in other CI?
 - [ ] Clean up and merge of Rust for powerpc support code

# Blockers for actual usage
 - [ ] Mixed Rust (LLVM) / GCC build supported ?
 - [ ] Distros switch to clang build ? (unlikely)
     - [ ] clang doesn't support `-fpatchable-function-entry` for ppc64le
 - [ ] GCC Rust support merged and available ?
 - [ ] ppc64le upgraded to Rust [Tier 1 status](https://doc.rust-lang.org/nightly/rustc/platform-support.html)
 - [ ] Distro support for Rust build (for some set of distros)

**Comment by mpe** [comment]
> > Clean up and merge of Rust for powerpc support code
> 
> Is that referring to bits in Rust-for-Linux/rust that isn't upstream yet?

Yes. It _may_ just be the `select HAVE_RUST` in `arch/powerpc/Kconfig`, but I haven't looked at the diff lately between Rust-for-Linux/rust and mainline.

**Comment by mkchauras** [comment]
Support for ppc64le and ppc32be has been sent out.
https://lore.kernel.org/linuxppc-dev/20260204210125.613350-1-mkchauras@gmail.com/

**Comment by mkchauras** [comment]
The discussion for adding a target specific for linux in rust is underway. 
For reference see:
https://lore.kernel.org/all/CANiq72nEam8n_daX5LyYrpH=i71k+pb+HLn6EEPdJJc-Zi9Q3A@mail.gmail.com/

The current patch will work fine from the support perspective. The target just makes it easier to do the same on the kernel side.

**Comment by mkchauras** [comment]
Latest upstream patch(Not merged yet)
https://lore.kernel.org/all/20260210090023.2587534-1-mkchauras@gmail.com/
